### ISSUE TITLE
Bug Report: vttestserver log files in $VTDATAROOT/logs are missing.

### ISSUE THREAD

**Issue by yoheimuta** [vitessio/vitess#15120]
### Overview of the Issue

With the recent vttestserver, we've identified that vttestserver log files in $VTDATAROOT/logs are missing.

Slack: https://vitess.slack.com/archives/C0PQY0PTK/p1706854811061539

### Reproduction Steps

1. Run the image

```
$ docker run --name=vttestserver \
  --platform linux/amd64 \
  --rm \
  -p 33577:33577 \
  --health-cmd="mysqladmin ping -h127.0.0.1 -P33577" \
  --health-interval=5s \
  --health-timeout=2s \
  --health-retries=5 \
  vitess/vttestserver:v18.0.2-ebf0106e \
  /vt/bin/vttestserver \
    --alsologtostderr \
    --port=33574 \
    --mysql_bind_host=0.0.0.0 \
    --keyspaces=test,unsharded \
    --num_shards=2,1 
```

2. Confirm a lack of log files

```
$ docker exec -it vttestserver /bin/bash

ls /vt/vtdataroot/vttest2314155067/logs/
```

### Binary Version

```sh
Versions with problems:
- Version: 19.0.0-SNAPSHOT (Git revision ebf0106e8e46ff0fb6ff0166ef9e9d79750be686, branch 'main'), built on Wed Jan 31 06:13:32 UTC 2024.
- Version: 18.0.2-SNAPSHOT (Git revision 2f43160047102ae728018ab03c61e0cfd9761033, branch 'release-18.0'), built on Mon Dec 18 10:15:38 UTC 2023.

Versions without problems:

- Version: 18.0.0-SNAPSHOT (Git revision 11264805ab1cabaa90642b35b801715ca673bc82, branch 'main'), built on Mon Sep 4 09:26:46 UTC 2023.
- Version: 17.0.1 (Git revision a4e751f8c28498a2aa4faebba3a83a8b684357af, branch 'HEAD'), built on Thu Aug 17 07:58:49 UTC 2023.
```

### Operating System and Environment details

```sh
Docker on M2 Mac
```

### Log Fragments

```sh
12:51:32 ~/vitessio/vitess $ docker exec -it vttestserver /bin/bash                                                                                          git/main
vitess@3f7dfd433ff3:/$ ls /vt/vtdataroot/vttest2314155067/
logs/          vt_0000000001/
vitess@3f7dfd433ff3:/$ ls /vt/vtdataroot/vttest2314155067/logs/
vitess@3f7dfd433ff3:/$ date; ls /vt/vtdataroot/vttest2314155067/logs/
Fri Feb  2 03:52:25 UTC 2024
vitess@3f7dfd433ff3:/$ exit;
exit
```

**Comment by deepthi** [comment]
@yoheimuta is it possible for you track down the commit where this behavior changed?

**Comment by yoheimuta** [comment]
@deepthi I've discovered that running the vttestserver binary directly on my M2 Mac doesn't cause the issue, regardless of whether it's version v18.0.0, v18.0.1, v18.0.2, or v19.0.0-SNAPSHOT ebf0106e8e46ff0fb6ff0166ef9e9d79750be686 (the latest main).

Therefore, the issue seems specific to running the binary in Docker on an M2 Mac. I've updated the `Operating System and Environment details` section of this issue accordingly. I'm in the process of identifying the problematic commit, but it may take some time since it involves building a Docker image.

**Comment by mattlord** [comment]
Thanks, @yoheimuta ! So perhaps the current implementation and/or docs assume that the files can be accessed via the local filesystem? With Docker on macOS a LinuxKit VM is used via the xhyve hypervisor or the (newish) Virtualization framework and that assumption would break.

**Comment by frouioui** [comment]
I can trace back the apparition of this issue to commit 35dd1b429e21a1034cbde000b765d00ee9349238, which updates the version of `glog` from `v1.0.0` to `v1.1.2`. I will investigate what changed from the previous and current version. Below is a table of the bisection I performed.

| SHA | Date | Repro passes |
| ----- | ----- | ----- |
| [v18.0.0](https://github.com/vitessio/vitess/releases/tag/v18.0.0) | Nov 6 | ✅ |
| [5a3ca083dbd6167819a2344a34a17ae4777393c5](https://github.com/vitessio/vitess/commit/5a3ca083dbd6167819a2344a34a17ae4777393c5) | Nov 10 | ✅ |
| [7e90aa2540b25313d6fc37b5379a1752c1111873](https://github.com/vitessio/vitess/commit/7e90aa2540b25313d6fc37b5379a1752c1111873) | Nov 10 | ✅ |
| [35dd1b429e21a1034cbde000b765d00ee9349238](https://github.com/vitessio/vitess/commit/35dd1b429e21a1034cbde000b765d00ee9349238) | Nov 10 | ⛔️ |
| [v18.0.1](https://github.com/vitessio/vitess/releases/tag/v18.0.1) | Nov 22 | ⛔️ |
| [d76f47992aca9801858e8b54a6f6e5912c0239e3](https://github.com/vitessio/vitess/commit/d76f47992aca9801858e8b54a6f6e5912c0239e3) | Dec 5 | ⛔️ |
| [2f43160047102ae728018ab03c61e0cfd9761033](https://github.com/vitessio/vitess/commit/2f43160047102ae728018ab03c61e0cfd9761033) | Dec 15 | ⛔️ |

**Comment by frouioui** [comment]
@yoheimuta, I am still looking on finding a solution to the issue. In the meantime you can see all the logs in `ls tmp/` if you are on Docker, on Mac it would be `ls $TMPDIR/`.

```
vitess@35eecd00828c:/$ ls tmp/
mysqlctl.35eecd00828c.vitess.log.ERROR.20240207-220021.18    vtcombo.ERROR
mysqlctl.35eecd00828c.vitess.log.INFO.20240207-220021.18     vtcombo.INFO
mysqlctl.35eecd00828c.vitess.log.WARNING.20240207-220021.18  vtcombo.WARNING
mysqlctl.ERROR                                               vttestserver.35eecd00828c.vitess.log.ERROR.20240207-220021.1
mysqlctl.INFO                                                vttestserver.35eecd00828c.vitess.log.INFO.20240207-220021.1
mysqlctl.WARNING                                             vttestserver.35eecd00828c.vitess.log.WARNING.20240207-220021.1
vtcombo.35eecd00828c.vitess.log.ERROR.20240207-220024.715    vttestserver.ERROR
vtcombo.35eecd00828c.vitess.log.INFO.20240207-220024.715     vttestserver.INFO
vtcombo.35eecd00828c.vitess.log.WARNING.20240207-220024.715  vttestserver.WARNING
```

**Comment by frouioui** [comment]
After forking the `glog` Go package and modifying the code that handles the [creation of the log files](https://github.com/frouioui/glog/blob/c97c4c6ecffc9fe846d98bf62fc7faa78f618340/glog_file.go#L121-L148), I am able to see that locally `logDirs` (which is a slice containing the value of the `--log_dir` and `/tmp`) is set to:

```
panic: /Users/florentpoinsard/Code/vitess/vtdataroot/vttest3115060181/logs,/var/folders/gh/3x61bzxj00ncmq5_j8wnqy2w0000gn/T
```

We can see the logs folder inside VTDATAROOT first, followed by my Mac's temporary directory (`$TMPDIR`). On the other hand, when running inside the Docker container, here is what we see:
```
panic: /tmp
```

We can only see the `/tmp` directory, not the VTDATAROOT's logs folder. Even though the `--log_dir` flag is properly set when starting `vtcombo`:
https://github.com/vitessio/vitess/blob/574162f4cad770c19a20a28526fe46f7de24b999/go/vt/vttest/vtprocess.go#L133

**Comment by frouioui** [comment]
Hello @yoheimuta! This bug should now be fixed, the fix has been merged on `release-18.0` already: https://github.com/vitessio/vitess/pull/15185.

**Comment by yoheimuta** [comment]
@frouioui Thank you for the fix 👍
